### ISSUE TITLE
Improvements in 0.5.1

### ISSUE THREAD

**Issue by abacao** [nukeop/nuclear#426]
Hello, once again congrats on the new release.

Current system: Ubuntu 19.04 win i3WM. deb version installed.

Some suggestions on the new release:

**1 - "back button":**
***Current behaviour***
    a) if "Shuffle on previous track" ON, the player will shuffle when going back.
    b) if "Shuffle on previous track" OFF, the player will start playing the previous song on the queue.
***Behaviour expected.***
    c) if "Shuffle on previous track" ON, the player will play the previously played music. 

**Example:**
**Story**: *I have the player with  "Shuffle ON" and I have been listening to the music in the queue. 
I'm currently on `**music 4**` where the arrow is.*

Queue
```
 1  2  3  4  5
```

Flow/Music played/Order
```
 2  5  1 **-> music 4**  3
```

What happens today: 
  a) if "Shuffle on previous track" ON, the player will shuffle when going back. Ex: goes to the `music 2` because it shuffles when going back.
  b) if "Shuffle on previous track" OFF, the player will start playing the previous song on the queue. Ex: goes to the `music 3`

What should happen in the future:
  c) if "Shuffle on previous track" ON, the player should go back in music played. Ex: should go to the `music 1`

To achieve this a "history" of the played music should be created in order to be able to go back several times.

Also, "history" could give you the possibility to keep some "insights". You could create a TOP10 of played music, the user could "fly back in time aka TimeMachine" and listen to the same music he/she hear "last week, last month" and with that create a TOP20/50 of all-time favorites:tada:

**2 - "Minimize window/small player.":**
When minimizing the window, I can get this:
[![https://imgur.com/7GDcq3s.png](https://imgur.com/7GDcq3sl.png)](https://imgur.com/7GDcq3sl.png)

but I found some issues: 
[![https://imgur.com/9hq2vgd.png](https://imgur.com/9hq2vgdl.png)](https://imgur.com/9hq2vgdl.png)

**Improvement suggestions:**
:point_up: 
1. If the window is small, the image could also be resized
2. The title and band, if it cannot be displayed, the items could be in a marquee/scrolling text.
3. The control elements could also be resizable (see "next track" button).
4. The volume could also be smaller (maybe leave on only the icon and "if on top of volume icon, use the scroll wheel to change volume").
5. If I scroll in that window, I get the rest of the player (doesn't really make sense).
[![https://imgur.com/kzvvDhp.png](https://imgur.com/kzvvDhpl.png)](https://imgur.com/kzvvDhpl.png)

**3 - "Nuclear started in the command line.":**
```bash
❯ nuclear       
renderer › {}
renderer › {}
renderer › {}
renderer › {}
renderer › {}
renderer › {}
```

Can this have more informative information? Info on the current music, action on the player (next, back, resize, clicked favorite, get lyrics, music URL, etc...)

**4 - "API.":**
Cannot see how to use the API:
[![https://imgur.com/ogAuyTM.png](https://imgur.com/ogAuyTMl.png)](https://imgur.com/ogAuyTMl.png)

Unless there is a valid reason, the API option should be disabled by default due to security reasons. 

**5 - "Github.":**
Was able to login to Github but nothing happens after that.
Suggestion: remove the `Github` section until fully deployed.

**6 - "File system.":**
While exploring the folder of Nuclear, I've found:
[![https://imgur.com/Y4bjzdP.png](https://imgur.com/Y4bjzdP.png)](https://imgur.com/Y4bjzdP.png)

In Linux the capitalization matters so you may find some issues with this in the future.

[![https://imgur.com/rK4pdRN.png](https://imgur.com/rK4pdRN.png)](https://imgur.com/rK4pdRN.png)
Why do I have files for other OS in my installation? Please remove it.

**7 - "CHANGELOG.md.":**
I would like to see a file in the repo where I could see the changelog (more or less what you have been doing in https://github.com/nukeop/nuclear/releases)
example: https://github.com/olivierlacan/keep-a-changelog/blob/master/CHANGELOG.md

**Comment by nukeop** [comment]
Thanks for the suggestions. Looks like some things don't work the same way in dev/production environment (namely api docs).
Responsiveness will definitely need to be improved, but in the meantime somebody's working on the miniplayer feature which should work better for smaller sizes of the window.

**Comment by aviezab** [comment]
First loading the UI is broken like this.
Please fix in 0.5.1 too

![image](https://user-images.githubusercontent.com/34294278/63737447-a801cb80-c8b0-11e9-9449-a4f1773da891.png)
